pkg/cluster/deploystorage_resources.go

```diff
@@ -141,6 +141,30 @@ func (m *manager) networkPrivateLinkService(installConfig *installconfig.Install
 	}
 }
 
+func (m *manager) networkPrivateEndpoint() *arm.Resource {
+	return &arm.Resource{
+		Resource: &mgmtnetwork.PrivateEndpoint{
+			PrivateEndpointProperties: &mgmtnetwork.PrivateEndpointProperties{
+				Subnet: &mgmtnetwork.Subnet{
+					ID: to.StringPtr(m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID),
+				},
+				ManualPrivateLinkServiceConnections: &[]mgmtnetwork.PrivateLinkServiceConnection{
+					{
+						Name: to.StringPtr("gateway-plsconnection"),
+						PrivateLinkServiceConnectionProperties: &mgmtnetwork.PrivateLinkServiceConnectionProperties{
+							PrivateLinkServiceID: to.StringPtr("/subscriptions/" + m.env.SubscriptionID() + "/resourceGroups/" + m.env.GatewayResourceGroup() + "/providers/Microsoft.Network/privateLinkServices/gateway-pls-001"),
```

Comment:
// TODO: in the future we will need multiple pls's.  It will be necessary to somehow allocate which cluster gets linked to which pls somewhere around here